File: src/code/Utilisateur.java
import java.util.ArrayList;

import javafx.util.Pair;

/**
 * @file Ingredient.h
 * @author Chourreau Mathieu, Fermé Léo, Piel Nathan, Zaza Souleymen
 * @brief 
 * @version 0
 * @date 2023-11-16
 * 
 * @copyright Copyright (c) 2023
 * 
 */

public class Utilisateur
{
    //ATTRIBUTS
    private int identifiant;
    private String nom;
    private String prenom;
    private int budget;
    private int tempsMax;
    private ArrayList<Pair<Ingredient, Double>> mesPreferences;

    //ENCAPSULATION
    public int getIdentifiant(){
        return identifiant;
    }

    public void setIdentifiant(int i){
        identifiant=i;
    }

    public String getNom(){
        return nom;
    }

    public void setNom(String n){
        nom=n;
    }

    public String getPrenom(){
        return prenom;
    }

    public void setPrenom(String p){
        prenom=p;
    }

    public int getBudget(){
        return budget;
    }

    public void setBudget(int b){
        budget=b;
    }

    public int getTempsMax(){
        return tempsMax;
    }

    public void setTempsMax(int t){
        tempsMax = t;
    }

    public ArrayList<Pair<Ingredient, Double>> getMesPreferences(){
        return mesPreferences;
    }

    public boolean ajouterIngredient(Ingredient i, double p){
        Pair<Ingredient, Double> pairAjt = new Pair<Ingredient,Double>(i, p);
        return mesPreferences.add(pairAjt);
    }

    public boolean retirerIngredient(Ingredient i){
        for(Pair<Ingredient,Double> ing:getMesPreferences()){
            if (ing.getKey() == i) {
                return mesPreferences.remove(ing);
            }
        }
        return false;
    }

    public boolean existeIngredient(Ingredient i){
        for(Pair<Ingredient,Double> ing:getMesPreferences()){
            if (ing.getKey() == i) {
                return mesPreferences.contains(ing);
            }
        }
        return false;
    }
}

File: src/code/Utilisateur.php
<?php

class Utilisateur {
    private $identifiant;
    private $nom;
    private $prenom;
    private $budget;
    private $tempsMax;
    private $mesPreferences;

    public function getIdentifiant() {
        return $this->identifiant;
    }

    public function setIdentifiant($i) {
        $this->identifiant = $i;
    }

    public function getNom() {
        return $this->nom;
    }

    public function setNom($n) {
        $this->nom = $n;
    }

    public function getPrenom() {
        return $this->prenom;
    }

    public function setPrenom($p) {
        $this->prenom = $p;
    }

    public function getBudget() {
        return $this->budget;
    }

    public function setBudget($b) {
        $this->budget = $b;
    }

    public function getTempsMax() {
        return $this->tempsMax;
    }

    public function setTempsMax($t) {
        $this->tempsMax = $t;
    }

    public function getMesPreferences() {
        return $this->mesPreferences;
    }

    public function ajouterIngredient($i, $p) {
        $pairAjt = array($i, $p);
        return array_push($this->mesPreferences, $pairAjt);
    }

    public function retirerIngredient($i) {
        foreach ($this->getMesPreferences() as $ing) {
            if ($ing[0] == $i) {
                $index = array_search($ing, $this->mesPreferences);
                if ($index !== false) {
                    unset($this->mesPreferences[$index]);
                    return true;
                }
            }
        }
        return false;
    }

    public function existeIngredient($i) {
        foreach ($this->getMesPreferences() as $ing) {
            if ($ing[0] == $i) {
                return true;
            }
        }
        return false;
    }
}

?>